client/src/app/home/home/home.component.html

```diff
@@ -0,0 +1,75 @@
+<div class="main">
+  <div class="main__accounts">
+    <app-account-card
+      *ngFor="let account of accounts; let i = index; trackBy: trackBy"
+      [account]="account"
+    >
+    </app-account-card>
+  </div>
+
+  <div class="main__transactions">
+    <mat-form-field
+      class="main__transactions__searchfield"
+      appearance="outline"
+    >
+      <mat-label>Search</mat-label>
+      <input matInput type="text" [(ngModel)]="value" />
+      <button
+        *ngIf="value"
+        matSuffix
+        mat-icon-button
+        aria-label="Clear"
+        (click)="value = ''"
+      >
+        <mat-icon>close</mat-icon>
+      </button>
+    </mat-form-field>
+
+    <p
+      *ngIf="transactionService.transactions.length === 0"
+      class="main__notransactions"
+    >
+      You don't have any transactions in this account!
+    </p>
+    <app-transaction-card
+      *ngFor="
+        let transaction of transactionService.filteredTransactions
+          | searchFilter: value;
+        let i = index;
+        trackBy: trackBy
+      "
+      [transaction]="transaction"
+    ></app-transaction-card>
+  </div>
+
+  <div class="main__options">
+    <div class="filter-buttons">
+      <button
+        [ngClass]="{ selected: transactionService.isIncomeFilter }"
+        class="filter-buttons__income"
+        mat-mini-fab
+        aria-label="income"
+        (click)="transactionService.filterTransactions('income')"
```

Comment:
create function in your component, this function will call transactionService.filterTransactions('income').
transactionService.filterTransactions('income') should return array and you will save it in variable filteredTransactions in this component. Something like this